apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/functions/NeutronFloatingIpToFloatingIp.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.openstack.nova.v2_0.compute.functions;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+import org.jclouds.domain.Location;
+import org.jclouds.openstack.nova.v2_0.domain.FloatingIP;
+
+import com.google.common.base.Function;
+import com.google.inject.assistedinject.Assisted;
+
+public class NeutronFloatingIpToFloatingIp
+      implements Function<org.jclouds.openstack.neutron.v2.domain.FloatingIP, FloatingIP> {
+
+   public interface Factory {
+      NeutronFloatingIpToFloatingIp create(Location location);
+   }
+
+   private final Location location;
+
+   @Inject
+   public NeutronFloatingIpToFloatingIp(@Assisted Location location) {
+      this.location = location;
+   }
+
+   @Override
+   public FloatingIP apply(@Nullable org.jclouds.openstack.neutron.v2.domain.FloatingIP floatingIP) {
+      FloatingIP.Builder<?> builder = FloatingIP.builder();
+      builder.id(floatingIP.getId());
+      builder.ip(floatingIP.getFloatingIpAddress());
+      builder.fixedIp(floatingIP.getFixedIpAddress());
+      builder.instanceId(floatingIP.getPortId());
```

Comment:
Isn't `instanceId` the ID of the node? In that case, the value should be the `deviceId` of the port; not the id of the port itself.

Reply:
ouch this function is not used anyway, so I'll remove it